test/python/transpiler/test_enlarge_with_ancilla_pass.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""Test the enlarge-with-ancilla pass"""
+
+import unittest
+
+from qiskit import QuantumRegister, QuantumCircuit
+from qiskit.mapper import Layout
+from qiskit.transpiler.passes import EnlargeWithAncilla
+from qiskit.converters import circuit_to_dag
+from ..common import QiskitTestCase
+
+
+class TestEnlargeWithAncilla(QiskitTestCase):
+    """ Tests the EnlargeWithAncilla pass."""
+
+    def setUp(self):
+        self.qr3 = QuantumRegister(3, 'qr')
+        circuit = QuantumCircuit(self.qr3)
+        circuit.h(self.qr3)
+        self.dag = circuit_to_dag(circuit)
+
+    def test_no_extension(self):
+        """ There are no idle physical bits to extend."""
+        layout = Layout([(self.qr3, 0),
+                         (self.qr3, 1),
+                         (self.qr3, 2)])
+
+        pass_ = EnlargeWithAncilla(layout)
+        after = pass_.run(self.dag)
+
+        qregs = list(after.qregs.values())
+        self.assertEqual(1, len(qregs))
+        self.assertEqual(self.qr3, qregs[0])
+
+    def test_with_extension(self):
+        """ There are 2 idle physical bits to extend."""
+        layout = Layout([(self.qr3, 0),
+                         None,
+                         (self.qr3, 1),
+                         None,
+                         (self.qr3, 2)])
+
+        pass_ = EnlargeWithAncilla(layout)
+        after = pass_.run(self.dag)
+
+        qregs = list(after.qregs.values())
+        self.assertEqual(2, len(qregs))
+        self.assertEqual(self.qr3, qregs[0])
+        self.assertEqual(QuantumRegister(2, name='ancilla'), qregs[1])
+
+    def test_name_coalition(self):
+        """Name coalition during ancilla extension."""
+        qr_ancilla = QuantumRegister(3, 'ancilla')
+        circuit = QuantumCircuit(qr_ancilla)
+        circuit.h(qr_ancilla)
+        dag = circuit_to_dag(circuit)
+
+        layout = Layout([(qr_ancilla, 0),
+                         None,
+                         (qr_ancilla, 1),
+                         None,
+                         (qr_ancilla, 2)])
+
+        pass_ = EnlargeWithAncilla(layout)
+        after = pass_.run(dag)
+
+        qregs = list(after.qregs.values())
+        self.assertEqual(2, len(qregs))
+        self.assertEqual(qr_ancilla, qregs[0])
+        self.assertEqual(2, qregs[1].size)
+        self.assertTrue(qregs[1].name.startswith('ancilla'))
```

Comment:
Testing that `qregs[1].name` equals `ancilla0`, or matches `^ancilla\d+$` might be clearer, since`ancilla` also `.startswith('ancilla')`.

Reply:
Fixed in 2f8b5deb